### ISSUE TITLE
Saving model

### ISSUE THREAD

**Issue by marceloclribeiro** [mazurowski-lab/segmentation-guided-diffusion#23]
Hello Nicholas! First of all thanks for the repo.
After changing the code to fix the previous informed bug of saving the model metioned in the Troubleshooting/Bugfixing section, im having the following problem on saving the model:

`self.eval_dataloader=<torch.utils.data.dataloader._SingleProcessDataLoaderIter object at 0x78ebf2971eb0> of type <class 'torch.utils.data.dataloader._SingleProcessDataLoaderIter'> cannot be saved.
self.external_config=TrainingConfig(model_type='DDIM', image_size=32, train_batch_size=16, eval_batch_size=2, num_epochs=1000, gradient_accumulation_steps=1, learning_rate=2e-05, lr_warmup_steps=500, save_image_epochs=10, save_model_epochs=10, mixed_precision='fp16', output_dir='ddim-figshare-32-segguided', push_to_hub=False, hub_private_repo=False, overwrite_output_dir=True, seed=0, segmentation_guided=True, segmentation_channel_mode='single', num_segmentation_classes=2, use_ablated_segmentations=False, dataset='figshare', resume_epoch=None, class_conditional=False, cfg_p_uncond=0.2, cfg_weight=0.3, trans_noise_level=0.5, use_cfg_for_eval_conditioning=True, cfg_maskguidance_condmodel_only=True) of type <class 'training.TrainingConfig'> cannot be saved.
`

Any tips on how to fix it?
